integrated_channels/logger.py

```diff
@@ -0,0 +1,128 @@
+import json
+import logging
+import sys
+from datetime import datetime
+from django.conf import settings
+from utils import generate_formatted_log
+
+USE_JSON_LOGGING = getattr(settings, 'INTEGRATED_CHANNELS_JSON_LOGGING', False)
+
+
+class IntegratedChannelsFormatter(logging.Formatter):
+    """
+    Custom formatter for integrated channels that supports both JSON and string formats.
+    """
+
+    def __init__(self):
+        self.use_json = USE_JSON_LOGGING
+        super().__init__()
+
+    def format(self, record):
+        if self.use_json:
+            log_entry = {
+                'timestamp': datetime.fromtimestamp(record.created).strftime('%Y-%m-%d %H:%M:%S,%f')[:-3],
```

Comment:
How about making the timestamp ISO 8601 compliant?

Reply:
We can, but we need to make it backward compatible; that's why I used the same format as before